eventdata/parameter_sources/randomevent.py

```diff
@@ -295,6 +295,11 @@ def __init__(self, params, agent=Agent, client_ip=ClientIp, referrer=Referrer, r
         self.total_days = params.get("number_of_days")
         self.remaining_days = self.total_days
         self.record_raw_event_size = params.get("record_raw_event_size", False)
+        self._offset = 0
+        self._web_host_index = 0
```

Comment:
is it worth using `itertools.cycle()` changing this to:

```
import itertools

...

self._web_host_index = itertools.cycle([1, 2, 3])
```

and then remove 339: `        self._web_host_index = (self._web_host_index + 1) % 3`

and replace 340:

`event["hostname"] = "web-%s-%s.elastic.co" % (event["geoip_continent_code"], self._web_host_index)`

with

`event["hostname"] = "web-%s-%s.elastic.co" % (event["geoip_continent_code"], self._web_host_index.cycle())`

My thinking is that it may be easier to figure things out in the future and that [itertools](https://docs.python.org/3/library/itertools.html) code claims to be optimized in speed and memory.

Reply:
I did a [benchmark](https://gist.github.com/danielmitterdorfer/09568ae70bcb7e63f0ca8a4649fd61e8) and it's faster indeed:

```
-------------------------------------------------------------------------------------- benchmark 'iteration': 2 tests --------------------------------------------------------------------------------------
Name (time in ns)                 Min                   Max                Mean              StdDev              Median                IQR              Outliers  OPS (Mops/s)            Rounds  Iterations
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
test_itertools               160.6900 (1.0)      1,160.4700 (1.0)      172.8763 (1.0)       44.9199 (1.0)      162.9500 (1.0)      11.1799 (5.44)      1912;1988        5.7845 (1.0)       61634         100
test_plain_old_iteration     266.6105 (1.66)     8,637.3887 (7.44)     285.2045 (1.65)     118.9893 (2.65)     271.3334 (1.67)      2.0559 (1.0)      2233;28489        3.5063 (0.61)     187759          18
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

This is not at all our bottleneck though, I did this mostly out of curiosity. However, I think it makes the code also a bit cleaner so I've pushed https://github.com/elastic/rally-eventdata-track/pull/70/commits/0e397dce0916aac4a2c35565da6ccde1416da019 which switches to `itertools.cycle()`.

Reply:
My main motivation was to make this easier to understand down the road as long provided it's not slower. Thanks for verifying and adding.

Reply:
Thanks for the suggestion! :)